src/Data/ByteString/Base64/URL.hs

```diff
@@ -91,13 +93,13 @@ encodeBase64' = encodeBase64_ base64UrlTable
 -- >>> decodeBase64 "PDw-Pg"
 -- Right "<<>>"
 --
-decodeBase64 :: ByteString -> Either Text ByteString
-decodeBase64 bs@(PS _ _ !l)
-    | l == 0 = Right bs
-    | r == 0 = unsafeDupablePerformIO $ decodeBase64_ decodeB64UrlTable bs
-    | r == 2 = unsafeDupablePerformIO $ decodeBase64_ decodeB64UrlTable (BS.append bs "==")
-    | r == 3 = validateLastPad bs $ decodeBase64_ decodeB64UrlTable (BS.append bs "=")
-    | otherwise = Left "Base64-encoded bytestring has invalid size"
+decodeBase64 :: Base64 'UrlPadded ByteString -> Either Text ByteString
```

Comment:
For URL-safe alphabets, this should be a union of `Base64 ('UrlPadded \/ 'UrlPadded)`

Reply:
Resolved in e2bdbe1